engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -342,10 +342,11 @@ public Answer sendTo(final Long dcId, final HypervisorType type, final Command c
                 }
                 Answer answer = null;
                 try {
-
-                    final long targetHostId = _hvGuruMgr.getGuruProcessedCommandTargetHost(host.getId(), cmd);
+                    final long targetHostId = _hvGuruMgr.getGuruProcessedCommandTargetHost(host.getId(), cmd, host.getHypervisorType());
                     answer = easySend(targetHostId, cmd);
                 } catch (final Exception e) {
+                    s_logger.error("Error sending command " + cmd.getClass().getName() + " to host: " +
+                            host.getUuid(), e);
```

Comment:
please do not include the stacktrace on error level, on debug level is good enough. Most operators will not have a use for it and only be alarmed. 
```suggestion
                    s_logger.error(String.format("Error sending command %s to host %s, due to %s", cmd.getClass().getName(),
                            host.getUuid(), e.getLocalizedMessage()));
                    if (s_logger.isDebugEnabled()) {
                        s_logger.debug(String.format("Error sending command %s" + " to host %s",
                            cmd.getClass().getName(),
                            host.getUuid(), e));
                    }
```

Reply:
Done, thanks